### ISSUE TITLE
[v0.8-beta.3] Multiple ingresses with mixed non-SSL / SSL rules not working

### ISSUE THREAD

**Issue by tim-smart** [jcmoraisjr/haproxy-ingress#407]
This is causing issues with cert-manager's ACME HTTP-01 verification.

The cert-manager ingress:
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    nginx.ingress.kubernetes.io/whitelist-source-range: 0.0.0.0/0,::/0
  creationTimestamp: "2019-10-01T01:58:52Z"
  generateName: cm-acme-http-solver-
  generation: 1
  labels: [redacted]
  name: cm-acme-http-solver-k2744
  namespace: contentful-proxy
  ownerReferences:
  - apiVersion: certmanager.k8s.io/v1alpha1
    blockOwnerDeletion: true
    controller: true
    kind: Challenge
    name: contentful-proxy-tls-907891456-4
    uid: 01ee70a5-e3ef-11e9-a0c0-42010a980158
  resourceVersion: "160067692"
  selfLink: /apis/extensions/v1beta1/namespaces/contentful-proxy/ingresses/cm-acme-http-solver-k2744
  uid: 04a41d4f-e3ef-11e9-a0c0-42010a980158
spec:
  rules:
  - host: [redacted-domain-1]
    http:
      paths:
      - backend:
          serviceName: cm-acme-http-solver-tbj2h
          servicePort: 8089
        path: /.well-known/acme-challenge/HNFztZBaMSqFmFbRSx95cyBUDUgcAJyZbBRtRn9cRHg
```

Actual ingress:
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    ingress.kubernetes.io/force-ssl-redirect: "true"
    kubectl.kubernetes.io/last-applied-configuration: [redacted]
    kubernetes.io/tls-acme: "true"
  creationTimestamp: "2019-10-01T01:58:40Z"
  generation: 1
  name: contentful-proxy-ing
  namespace: contentful-proxy
  resourceVersion: "160071087"
  selfLink: /apis/extensions/v1beta1/namespaces/contentful-proxy/ingresses/contentful-proxy-ing
  uid: fd321767-e3ee-11e9-a0c0-42010a980158
spec:
  rules:
  - host: [redacted-domain-1]
    http:
      paths:
      - backend:
          serviceName: contentful-ariseconf19
          servicePort: 80
    [...]
  tls:
  - hosts:
    - [redacted-domain-1]
    - [redacted-domain-2]
    - [redacted-domain-3]
    - [redacted-domain-4]
    - [redacted-domain-5]
    secretName: contentful-proxy-tls
```

By curl-ing the path `/.well-known/acme-challenge/HNFztZBaMSqFmFbRSx95cyBUDUgcAJyZbBRtRn9cRHg` I discovered that it was hitting the default backend instead of the http solver pod.

Downgrading to 0.7.x fixed the issue.

**Comment by jcmoraisjr** [comment]
Hi, there was an issue matching paths with lower and upper chars on :80/http. This should be fixed in v0.8-beta.4.

**Comment by jcmoraisjr** [comment]
Closing. Please update this issue if you find any problem.
